Functions/xASL_io_ExportVTK.m

```diff
@@ -0,0 +1,133 @@
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:          To export the "test" NIFTI to structured points in VTK format, you can run the following lines.
+%                   [x] = ExploreASL_Initialize('',0);
+%                   nifti = '.\test.nii';
+%                   xASL_io_ExportVTK(x.MyPath,nifti);
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Check ExploreASL
+    if nargin < 1
+        warning('Could not find ExploreASL...');
+        return 
+    end
+
+    %% Load image
+    if nargin < 2
+       warning('Neither input image path nor matrix given...');
+       return 
+    end
+    
+    if ischar(nifti)
+        % Import NIFTI
+        image = xASL_io_Nifti2Im(nifti);
+    elseif isnumeric(nifti)
+        % Import image
+        image = nifti;
+    end
+    
+    % Check dimensions
+    if numel(size(image))~=3 && numel(size(image))~=4
+        error('Input image dimensions are incorrect...');
+    end
+    if numel(size(image))==4
+        fprintf('4D image will be exported as a VTK time series...\n');
+    end
+
+    % Apply mask (Only supported for 3D images right now)
+    if numel(size(image))==3
+        if nargin > 2 && exist('mask','var')
+            if ~isempty(mask)
+                % Get mask
+                imageMask = xASL_io_Nifti2Im(mask);
+                % Resample mask to image
+                % [maskResampled] = xASL_im_ResampleLinearFair(imageMask,size(image)); % Old
+                [maskResampled] = xASL_im_ResampleIM(imageMask, [1 0 0 0;0 1 0 0;0 0 1 0; 0 0 0 1], [1 0 0 0;0 1 0 0;0 0 1 0; 0 0 0 1], size(image)); % New
+                fprintf('We assume that image and mask have the same orientation...\n');
+                % Apply mask
+                image = image.*maskResampled;
+            end
+        end
+    end
+    
+    % Check export path
+    if nargin < 4 && ~exist('exportPath','var')
+        if ischar(nifti)
+            % Define export file
+            folder = fileparts(nifti);
+            if numel(size(image))==3 % 3D
+                exportPath = fullfile(folder,'export.vtk');
+            else % 4D
+                exportPath = fullfile(folder,'export-NUM.vtk');
+            end
+        else
+            % Neither export path nor nifti path given
+            if numel(size(image))==3 % 3D
+                exportPath = fullfile(pwd,'export.vtk');
+            else % 4D
+                exportPath = fullfile(pwd,'export-NUM.vtk');
+            end
+        end
+    end
+    
+    % Select export method
+    fprintf('Export vtk as structured points...\n');
+    
+    % Export to vtk (MIT-License) % https://www.mathworks.com/matlabcentral/fileexchange/47814-vtkwrite-exports-various-2d-3d-data-to-paraview-in-vtk-file-format
+    readAndPrintTextFile(fullfile(pathExploreASL,'External','vtkwrite','MIT_License.txt'));
```

Comment:
I think that it is important to have this MIT-license file there. But we probably don't need to print it during the execution. At least - we don't do it for other functions.
1. I would thus remove this print and remove the pathExploreASL parameter.
2. We can mention in the header of the function here, that we use vtkwrite that has MIT license and mention the developer name. And we can print a single line about it here, but the whole license file probably not.

Reply:
Done in 0eeb1c7